### ISSUE TITLE
[camelcase] false negative in destructuring with assignment

### ISSUE THREAD

**Issue by yeonjuan** [eslint/eslint#13025]
**Tell us about your environment**

* **ESLint Version:** 7.0.0-alpha.2
* **Node Version:** v11.15.0
* **npm Version:** 6.7.0

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

* [Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgY2FtZWxjYXNlOiBbXCJlcnJvclwiXSAqL1xudmFyIG9iaiA9IHt9O1xuXG5vYmouZm9fbyA9IGQ7XG4oeyBhOiBvYmouZm9fbyB9ID0gb2JqKTtcbihbb2IuZm9fb10gPSBiYXIpO1xuXG4iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjExLCJzb3VyY2VUeXBlIjoibW9kdWxlIiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7ImJyb3dzZXIiOmZhbHNlLCJub2RlIjpmYWxzZX19fQ==)

```js
/*eslint camelcase: ["error"] */
var obj = {};

obj.fo_o = d;  // Warning
({ a: obj.fo_o } = {d: d}); // No Warning
([ob.fo_o] = bar); // No Warning
```

**What did you expect to happen?**

I expect 3 warnings.

**What actually happened? Please include the actual, raw output from ESLint.**

One warning

The below cases are not warned

```js
({ a: obj.fo_o } = {d: d}); // No warning
([ob.fo_o] = bar); // No Warning
```

**Are you willing to submit a pull request to fix this bug?**

Yes!

I think the proper behavior should be as belows.

```js
/*eslint camelcase: ["error"] */
obj.fo_o = d; // warning
({ a: obj.fo_o } = bar); // warning
([ob.fo_o] = bar); // warning
```

```js
/*eslint camelcase: ["error", {"ignoreDestructuring": true}] */
obj.fo_o = d; // warning
({ a: obj.fo_o } = bar); // warning
([ob.fo_o] = bar); // warning
```

```js
/*eslint camelcase: ["error", {"properties": "never"}] */
obj.fo_o = d; // No warning
({ a: obj.fo_o } = {d: d}); // No warning
([ob.fo_o] = bar); // No warning
```

**Comment by mdjermanovic** [comment]
This looks like a bug to me, too.

Regarding `ignoreDestructuring`, I think it's only for shorthand variable initialization/assignments and in my opinion these should be errors:

```js
/*eslint camelcase: ["error", {"ignoreDestructuring": true}] */

({ a: obj.fo_o } = bar); // error
([ob.fo_o] = bar); // error
```

**Comment by yeonjuan** [comment]
@mdjermanovic 
> Regarding ignoreDestructuring, I think it's only for shorthand variable initialization/assignments and in my opinion these should be errors:

Okay, I agreed. I changed the examples in the issue. Thanks.
